topic-operator/src/main/java/io/strimzi/operator/topic/v2/BatchingLoop.java

```diff
@@ -0,0 +1,309 @@
+
+    /**
+     * A thread that services the head of the {@link #queue}.
+     */
+    class LoopRunnable extends Thread {
+
+        private volatile boolean stopRequested = false;
+
+        LoopRunnable(String name) {
+            super(name);
+            setDaemon(false);
+        }
+
+        static final ReconciliationLogger LOGGER = ReconciliationLogger.create(LoopRunnable.class);
+        private volatile long lastLoop = System.nanoTime();
+
+        long msSinceLastLoop() {
+            return (System.nanoTime() - lastLoop) / 1_000_000;
+        }
+
+        @Override
+        public void run() {
+            LOGGER.debugOp("Entering run()");
+            Batch batch = new Batch(maxBatchSize);
+            int batchId = 0;
+            lastLoop = System.nanoTime();
+            while (!runOnce(batchId, batch)) {
+                batchId++;
+                lastLoop = System.nanoTime();
+            }
+            LOGGER.debugOp("Exiting run()");
+        }
+
+        public void requestStop() {
+            LOGGER.infoOp("Stop requested");
+            // In theory interrupting the thread should be enough to cause the InterruptedException
+            // to propagate up to runOnce(), but we can't completely guarantee that
+            // everything handles interruption properly, so we use stopRequested to ensure
+            // we'll at least stop after the return from stopOnce(), even if the exception gets swallowed
+            this.stopRequested = true;
+            this.interrupt();
+        }
+
+        private boolean runOnce(int batchId, Batch batch) {
+            try {
+                synchronized (BatchingLoop.this) {
+                    // remove the old batch from the inflight set and reset the batch
+                    LOGGER.debugOp("[Batch #{}] Removing batch from inflight", batchId - 1);
+                    batch.toUpdate.stream().map(TopicEvent::toRef).forEach(inFlight::remove);
+                    batch.toDelete.stream().map(TopicEvent::toRef).forEach(inFlight::remove);
+                    batch.clear();
+                    // fill a new batch
+                    fillBatch(batchId, batch);
+                }
+
+                if (batch.size() > 0) {
+                    LOGGER.infoOp("[Batch #{}] Reconciling {} topics", batchId, batch.size());
+                    // perform reconciliation on new batch
+                    if (!batch.toUpdate.isEmpty()) {
+                        controller.onUpdate(batch.toUpdate.stream().map(upsert -> lookup(batchId, upsert)).filter(Objects::nonNull).toList());
+                    }
+                    if (!batch.toDelete.isEmpty()) {
+                        controller.onDelete(batch.toDelete.stream().map(td -> new ReconcilableTopic(new Reconciliation("delete", "KafkaTopic", td.namespace(), td.name()), td.topic(), BatchingTopicController.topicName(td.topic()))).toList());
+                    }
+                    LOGGER.debugOp("[Batch #{}] Reconciled batch", batchId);
```

Comment:
this is at "DEBUG" but the corresponding "opening" log above is at "INFO" ... isn't it better to have them both at "DEBUG" to avoid polluting log on reconcile?